whatsapp_archive.py

```diff
@@ -159,7 +163,7 @@ def main():
     with open(args.input_file, 'rt', encoding='utf-8-sig') as fd:
         messages = IdentifyMessages(fd.readlines())
     template_data = TemplateData(messages, args.input_file)
-    HTML = FormatHTML(template_data)
+    HTML = ParseImg(FormatHTML(template_data))
```

Comment:
I know that you did what was easiest, but in the long run it will make the code hard to maintain. Instead of adding another step to the process, identification of images should happen in the IdentifyMessages function, and identified images should be added to the `messages` list returned by IdentifyMessages. Then in FormatHTML, there should be a line which looks at a message, and if it's an image, adds the `<img/>` tag around it.

This will involve changing line 82 to something like `messages.append((msg_date, msg_user, msg_body, msg_image))` where msg_image might be `None` if there was no image in the message. (I'm assuming only one image per message, is that correct?)

Please also look at the _test.py file, to add some tests to make sure that your code works now, and will work after the next person comes and makes their changes on top of yours.